### PR TITLE
Support for plugable reconcilers

### PR DESCRIPTION
**Pull request by p-strusiewiczsurmacki-mobica**
This PR introduces basic support for plugable reconcilers as it will be required by #99.

`LegacyReconciler` that uses netlink/FRR was separated from the `Reconciler` struct and is provided to this struct by `IReconciler` interface, via dependency injection. Additionally stub for NETCONF based reconciler was created and config flag for selection was added.

### CODE REVIEW

File: pkg/reconciler/reconciler.go [hunk 1]
```diff
@@ -3,92 +3,49 @@ package reconciler
...
+	getConfig() *config.Config
 }
 
-type reconcile struct {
-	*Reconciler
-	logr.Logger
+type Reconciler struct {
+	client      client.Client
+	logger      logr.Logger
+	iReconciler IReconciler
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/reconciler/reconciler.go]: Done

File: pkg/reconciler/reconciler.go [hunk 2]
```diff
@@ -18,32 +19,39 @@ import (
 
 const defaultDebounceTime = 20 * time.Second
 
+type Adapter interface {
```
Comment (schrej) [inline pkg/reconciler/reconciler.go:22]: If keep Adapter as the name here, implementations of it should also be called Adapter, not Reconciler.

**Review by schrej — CHANGES_REQUESTED** [review]

File: pkg/reconciler/reconciler.go (hunk 2 above)
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/reconciler/reconciler.go:22]: Sorry, fixed that.

**Comment by p-strusiewiczsurmacki-mobica** [comment]
Closing this as it will be addressed in #124

**Pull request closed without merging**